lib/api.js

```diff
@@ -972,6 +978,9 @@ module.exports = function(self, options) {
       // Modified. Could also be moved, so don't return
       if (!_.isEqual(value, toObjects.byId[value._id])) {
         if (description) {
+          // console.log(toObjects.dotPaths[value._id]);
+          // console.log(JSON.stringify(value, null, '  '));
+          // console.log(JSON.stringify(toObjects.byId[value._id], null, '  '));
```

Comment:
More debug comments here and above.

Reply:
These debug comments are likely to be useful a few more times before we see the last of debugging this algorithm, so I'm happy with leaving them present but commented like this.